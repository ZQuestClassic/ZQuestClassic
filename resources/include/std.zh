// std.zh
// 28th December, 2020 for ZC 2.55 Alpha 86
// Standard header for ZScript.
// Please don't edit this file. Instead, create a new header file for your
// quests' scripts,  add constants and procedures to that file,
// and #include it separately.

//Options
#option BINARY_32BIT off
	//Would break a number of things if set on
#option SHORT_CIRCUIT on
	//Should work for everything in std_zh, and just speed it up

#includepath "include"

//Version Metadata
#include "std_zh/std_meta.zh"

//User Configuration Files
#include "std_zh/std_user_defs.zh" //Typedefs: Always first.
#include "std_zh/std.cfg"

//Main Files
#include "std_zh/limits.zh"
#include "std_zh/std_constants.zh"
#include "std_zh/ffrules.zh"
#include "std_zh/std_functions.zh"
#include "std_zh/math.zh"
#include "std_zh/weapon.zh"
#include "std_zh/std_sideview.zh" //Sideview and platforming

//ZScript string handling functions, akin to those found in C
//#include "string.zh"
#include "std_zh/string/string_constants.zh" //Now #includeed from here. This prevents conflicts for those people
#include "std_zh/string/string_functions.zh" //who would #include string.zh from a secondary source, as that file is now 
			     //empty, and the string.zh components load from here. 

//Keyboard handling for 2.54 and above
#include "std_zh/std_keyboard.zh"

//Locale
#include "std_zh/locale/std_EnglishIntl.zh" //International English Localisation

//Legacy Wrappers
#include "std_zh/std_legacy.zh"