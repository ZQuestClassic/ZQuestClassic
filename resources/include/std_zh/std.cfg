//Settings

const int STD_FORCE_OLD_DISTANCE = 1;	//If enabled, Distance() will use Sqrt() instead of SafeSqrt().
					//! If you use this, disable the quest rule 'Log Script Errors to Allegro.log
					//! before distributing your quest!
					
const int STD_DISTANCE_USE_SAFESQRT = 0; 	//If enebaled, Distance will use SafeSqrt(). Otherwise, it will return a factor, 
					
const int STD_EMULATE_ENGINE_COLLISION = 0; // If enabled, Collision hitboxs will more precisely emulate the
					    // internal collision. This differs to how scripted collision
					    // has always worked.
					    

const int STD_CENTERXY_USES_HITBOXES = 0; // If enabled, CenterX() and CenterY() will use object hitboxes
					  // and offsets by default, rather than sprite sizes when calculating
					  // the centre of objects.
					  

const int STD_NEVER_USE_SAFESQRT = 1; //Enable to force any calls to SafeSqrt() to use Sqrt() instead.

const int MAX_HIGHEST_LEVEL_ITEM_CHECK = 255; //The highest item ID checked by GetHighestLevelItem(). 

const int REVISED_ROTATEPOINT_X_FUNCTION = 1; //Set to 0 to use 2.50.x RotatePointX()
const int REVISED_ROTATEPOINT_Y_FUNCTION = 1; //Set to 0 to use 2.50.x RotatePointY()


const int MAX_NOACTION_INPUT = 14; //Checks A, B, L, R, U/D/L/R, Ex1, Ex2, Ex3, Ex4
	//Use CB_MAX to force the analogue stick false. 
const int NOACTION_SKIP_MAP = 0; //If enabled, NoAction will not disable pressing the Map button
const int NOACTION_SKIP_START = 0; //If enabled, NoAction will not disable pressing the Start button

const int OLD_ONSIDEPLATFORM = 0; //Change to 1 to use old OnSidePlatform calls.

const int OLD_RAND_FUNCS = 0; //Change to 1 to use old Rand() logic
