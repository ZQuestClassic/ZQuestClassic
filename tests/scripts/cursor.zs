#include "std.zh"

ffc script Cursor
{
    void run(int combo)
    {
        while (true)
        {
			int x = Input->Mouse[MOUSE_X];
			int y = Input->Mouse[MOUSE_Y];
			Screen->DrawCombo(3, x, y,
				combo,
				1,
				1,
				3,
				-1, -1, 0, 0, 0, 0, 0, true, OP_OPAQUE);

            Waitframe();
        }
    }
}
