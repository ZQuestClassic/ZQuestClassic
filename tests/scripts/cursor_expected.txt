Compiling 'cursor.zs'
Pass 1: Parsing
Pass 2: Preprocessing
Pass 3: Registration
Pass 4: Analyzing Code
Pass 5: Checking code paths
Pass 6: Generating object code
Pass 7: Assembling
Success!
Compile finished with exit code '0' (success)

PUSHR d0; void run(int) Params
PUSHR REFFFC; void run(int) Body Start
PUSHARGSV 0,0.0002
SETR d4,SP
SETV d2,0; while(true) #8 Body [Opt:AlwaysOn] Start
PUSHR d2
POP d0
SETR d2,MOUSEARR
STORED d2,1
SETV d2,1
PUSHR d2
POP d0
SETR d2,MOUSEARR
STORED d2,0
PUSHV 3; InlineFunc[void Screen->DrawCombo(int, int, int, int, int, int, int, int, int, int, int, int, int, int, bool, int)] Params Start
LOADD d2,1
PUSHR d2
LOADD d2,0
PUSHR d2
LOADD d2,3
PUSHR d2
PUSHARGSV 1,0.0002
PUSHV 3
PUSHARGSV -1,0.0002
PUSHARGSV 0,0.0005
PUSHV 1
PUSHV 128; InlineFunc[void Screen->DrawCombo(int, int, int, int, int, int, int, int, int, int, int, int, int, int, bool, int)] Params End
DRAWCOMBO; InlineFunc[void Screen->DrawCombo(int, int, int, int, int, int, int, int, int, int, int, int, int, int, bool, int)] Body Start
POPARGS d5,0.0016; InlineFunc[void Screen->DrawCombo(int, int, int, int, int, int, int, int, int, int, int, int, int, int, bool, int)] Body End
WAITFRAME; InlineFunc[void Waitframe()] Body AND while(true) #8 Body [Opt:AlwaysOn] End
GOTO 5; while() #8 End AND [Opt:NeverRet] AND void run(int) Body End


ALLOCATEGMEMV d2,4096; void run() Body Start
SETR gd1,d2
QUIT; void run() Body End